mlrun/utils/helpers.py

```diff
@@ -667,6 +672,25 @@ def enrich_image_url(image_url: str, client_version: str = None) -> str:
     return image_url
 
 
+def _resolve_image_tag_suffix(
+    tag: str = None, client_python_version: str = None
+) -> str:
+    """
+    resolves what suffix should be appended to the image tag
+    :param tag: the image tag version
+    :param client_python_version: the client python version
+    :return: the suffix to append to the image tag
+    """
+    if not client_python_version or not tag:
+        return ""
+
+    # For mlrun 1.3.0, we decided to support mlrun runtimes images with both python 3.7 and 3.9 images.
+    # While the python 3.9 images will continue to have no suffix, the python 3.7 images will have a '-py37' suffix.
+    if tag == "1.3.0" and client_python_version < "3.9":
```

Comment:
👀 you expect `client_python_version` / `tag` to be a string and you use `<>` here?. use semver or anything compatible with parsing the python version in a correct way.
also, this will remain true for 1.3.x versions, at least. Id suggest changing the if to `>= 1.3.0`.
and make sure the image suffix is made of `-py{python_version.major}{python_version.minor}` for correctness.

Reply:
I would like to add that if the client python version is 3.8 we should have no suffix.
It will be safer to load the python 3.9 images instead of python 3.7 since the pickle protocol was changed in python 3.8